app/views/components/_contents-list.html.erb

```diff
@@ -16,20 +16,37 @@
       <%= t("content_item.contents") %>
     </h2>
     <ol class="app-c-contents-list__list">
-      <% contents.each do |contents_item| %>
+      <% contents.each_with_index do |contents_item, position| %>
```

Comment:
If you use .each.with_index(1) the index will start from 1 and you won't need to +1 to position later on in the logic. (same goes for nested position)